### PR TITLE
phone: Support TChannel

### PR DESCRIPTION
**Pull request by abhinav**
This is needed for context propagation testing.

### CODE REVIEW

**Comment by abhinav** [comment]
CC @yarpc/golang

**Comment by kriskowal** [comment]
word

File: crossdock/server/yarpc/phone.go [hunk 1]
```diff
@@ -53,9 +69,26 @@ type PhoneResponse struct {
...
+		url := fmt.Sprintf("http://%s:%d", t.Host, t.Port)
+		outbound = ht.NewOutbound(url)
+	case body.Transport.TChannel != nil:
+		t := body.Transport.TChannel
+		hostport := fmt.Sprintf("%s:%d", t.Host, t.Port)
+		ch, err := tchannel.NewChannel("yarpc-test-client", nil)
+		if err != nil {
+			return nil, nil, fmt.Errorf("failed to build TChannel: %v", err)
+		}
+
```
Comment (breerly) [inline crossdock/server/yarpc/phone.go]: nit - no need for 'dis space

**Comment by breerly** [comment]
lgtm

File: crossdock/server/yarpc/phone.go (hunk 1 above)
Comment (abhinav) [inline crossdock/server/yarpc/phone.go]: 👍

**Pull request merged**